pkg/controllers/user/rbac/legacy_grb_cleaner.go

```diff
@@ -0,0 +1,60 @@
+package rbac
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/rancher/norman/lifecycle"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type grbCleaner struct {
+	m *manager
+}
+
+func newLegacyGRBCleaner(m *manager) *grbCleaner {
+	return &grbCleaner{m: m}
+}
+
+func (p *grbCleaner) sync(key string, obj *v3.GlobalRoleBinding) (runtime.Object, error) {
```

Comment:
nit: I feel like some of this logic could be repackaged and reused for both, otherwise looks fine

Reply:
While I think there is some repetition here I don't think its enough to warrant attempting to reduce the overlap. While the logic is similar the signatures provide some helpful type information.